### PR TITLE
[Serve] log deployment success message in serve.run before blocking

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

In the previous PR, I accidentally blocked the cli logger from logging the deployment success message. log deployment success message in serve.run before blocking.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by GeneDer** [comment]
Also tested manually and seeing "Deployed app successfully." is logged again
image

**Comment by GeneDer** [comment]
> @GeneDer I don't really want to add a callback to the public API (to reduce the API surface to maintain). It's probably useful to log this when folks call `serve.run` too, how about we just add a logging statement there by default?

Partial bc there are 2 separate messages that we log "Deployed app successfully." and "Redeployed app successfully." and also the cli logger works different than our serve logger with some custom formatting and stuff. But yea let me simply by just always log a "Deployed app successfully." and drop the callback

**Comment by edoakes** [comment]
> > @GeneDer I don't really want to add a callback to the public API (to reduce the API surface to maintain). It's probably useful to log this when folks call `serve.run` too, how about we just add a logging statement there by default?
> 
> Partial bc there are 2 separate messages that we log "Deployed app successfully." and "Redeployed app successfully." and also the cli logger works different than our serve logger with some custom formatting and stuff. But yea let me simply by just always log a "Deployed app successfully." and drop the callback

Got it, the original motivation makes sense. Let's just simplify and go w/ the single message.

**Comment by GeneDer** [comment]
@edoakes this is how it looks now. It's no longer green and doesn't have the file name and line number. Also, noticed this is not using `ray.serve` logger. Do you think it makes sense to reuse serve logger for this entire file? 
image

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -573,14 +573,15 @@ def run(
...
+    logger.warning("Deployed app successfully.")
 
     if blocking:
         try:
             while True:
                 # Block, letting Ray print logs to the terminal.
                 time.sleep(10)
         except KeyboardInterrupt:
-            logger.info("Got KeyboardInterrupt, release blocking...")
+            logger.warning("Got KeyboardInterrupt, release blocking...")
```
Comment (edoakes) [inline python/ray/serve/api.py:584]: What does "release blocking" mean here?

Comment (edoakes) [reply python/ray/serve/api.py:584]: Maybe better to say "Got KeyboardInterrupt, exiting"

File: python/ray/serve/api.py [hunk 2]
```diff
@@ -573,14 +573,15 @@ def run(
         route_prefix=route_prefix,
         logging_config=logging_config,
     )
+    logger.warning("Deployed app successfully.")
```
Comment (edoakes) [inline python/ray/serve/api.py:576]:
```suggestion
    logger.warning(f"Deployed app '{name}' successfully.")
```

Comment (edoakes) [reply python/ray/serve/api.py:576]: this should be `INFO`-level

**Comment by edoakes** [comment]
> @edoakes this is how it looks now. It's no longer green and doesn't have the file name and line number. Also, noticed this is not using `ray.serve` logger. Do you think it makes sense to reuse serve logger for this entire file? image

Do you know why it doesn't have any log line formatting like `INFO api.py ...` even though you're using the logger? That part looks out of place.

File: python/ray/serve/api.py (hunk 2 above)
Comment (GeneDer) [reply python/ray/serve/api.py:576]: Due to this not using serve logger, info level won't log it to stdout 😅

**Comment by GeneDer** [comment]
> > @edoakes this is how it looks now. It's no longer green and doesn't have the file name and line number. Also, noticed this is not using `ray.serve` logger. Do you think it makes sense to reuse serve logger for this entire file? image
> 
> Do you know why it doesn't have any log line formatting like `INFO api.py ...` even though you're using the logger? That part looks out of place.

Yes, bc it's not using serve's logger, it's doing `logger = logging.getLogger(__file__)`😅

**Comment by edoakes** [comment]
Oh I see @GeneDer. What does it look like if we use the serve logger? I think a lot of the fields we usually set there aren't applicable on the driver...

Maybe we should have an API logger (could maybe just pull in the `cli_logger` that's used in the `CLI` code).

**Comment by GeneDer** [comment]
This is how it looks if we just use serve logger (I didn't change back to info, but can easily do that)
image

**Comment by GeneDer** [comment]
@edoakes This is how it looks with the existing change
image

**Review by edoakes — APPROVED** [review]

**Pull request merged**